src/view/components/app.ts

```diff
@@ -22,33 +19,49 @@ import {HttpRequestUtils} from '../../api/util/httpRequestUtils';
  */
 export class AppComponent {
 
-    login: boolean= true;
+    login: boolean = true;
 
     // We are initializing dotcmsConfig in this component because it's the entry point and we need it
     // ready for main-component, maybe we can do the request if the login it's success
-    constructor(private router: Router, private loginService: LoginService, dotcmsConfig: DotcmsConfig, private httpRequestUtils: HttpRequestUtils) {}
+    constructor(private dotcmsConfig: DotcmsConfig, private loginService: LoginService, private router: Router) {
+    }
 
     ngOnInit(): void {
-        // TODO: change the httpRequestUtils.getQueryParams() with an NG2 method equivalent to QueryParams on NGRX.
-        let queryParams: Map = this.httpRequestUtils.getQueryParams();
+        let queryParams: Map = this.getQueryParams();
 
         if (<boolean> queryParams.get('resetPassword')) {
             let token: string = queryParams.get('token');
             let userId: string = queryParams.get('userId');
-            this.router.go(`public/resetPassword/${userId}?token=${token}`);
+            this.router.navigate([`public/resetPassword/${userId}?token=${token}`]);
         } else {
-            // TODO: not sure about this service returning and Observable or subscribe because only need this once.
-            this.loginService.loadAuth().subscribe(res => {
-                if (this.loginService.auth.user) {
-                    this.router.go('dotCMS');
-                } else {
-                    this.router.go('public/login');
-                }
+            // TODO: need to change this to a promise or maybe unify this into one service
+            this.dotcmsConfig.getConfig().subscribe(() => {
+                this.loginService.loadAuth().subscribe(() => {
+                    if (this.loginService.auth.user) {
+                        this.router.navigate(['dotCMS'])
+
+                    } else {
+                        this.router.navigate(['/public/login']);
+                    }
+                });
             });
+
         }
     }
 
     toggleMain(change: boolean): void {
         this.login = change;
     }
+
+    private getQueryParams(): Map<string, string> {
```

Comment:
Angular must have a better way to do this, ngrx had the follor:

https://github.com/ngrx/router/blob/master/docs/overview/params.md